### ISSUE TITLE
inlineSourceMap config conflicting with fallback svelte-preprocess config

### ISSUE THREAD

**Issue by trev-dev** [sveltejs/language-tools#1128]
Across 3 editors (VSCode, Emacs (lsp-mode) and Sublime Text (lsp-svelte)) I'm seeing this error in a Svelte component while set up with TypeScript: `<script lang="ts"> - [svelte-preprocess] Encountered type error`

It doesn't stop me from compiling or running the code, so it's a nuisance error, but it's a red squiggle nonetheless

System Info:

Arch Linux 5.10.54-1-lts
svelte-language-server@0.14.4

rollup:
```javascript
import ts from '@rollup/plugin-typescript'
import sveltePreprocess from 'svelte-preprocess'
import svelte from 'rollup-plugin-svelte'
import scss from 'rollup-plugin-scss'
import commonjs from '@rollup/plugin-commonjs'
import resolve from '@rollup/plugin-node-resolve'
import filesize from 'rollup-plugin-filesize'

import postcss from 'postcss'
import autoprefixer from 'autoprefixer'
import { terser } from 'rollup-plugin-terser'
import chalk from 'chalk'
import version from 'rollup-plugin-version-injector'

const prod = !process.env.ROLLUP_WATCH
console.log(`Currently running in ${
  prod ? chalk.green('production') : chalk.red('development')
}\n`)

const cssOptions = {
  output: 'assets/bundle.css'
}

export default {
  input: 'src/index.ts',
  output: {
    file: 'assets/bundle.js',
    format: 'iife',
    name: 'bundle',
    sourcemap: prod ? false : 'inline'
  },
  plugins: [
    svelte({
      preprocess: sveltePreprocess(),
      compilerOptions: {
        hydratable:  true,
        dev: !prod
      }
    }),
    ts({
      inlineSources: !prod,
      inlineSourceMap: !prod
    }),
    scss(!prod && { processor: () => postcss([autoprefixer()]) }),
    css(cssOptions),
    prod && terser(),
    commonjs(),
    resolve({browser: true}),
    prod && filesize(),
    prod && version()
  ],
  watch: {
    clearScreen: false
  }
}
```
tsconfig:
```javascript
{
  "extends": "@tsconfig/svelte/tsconfig.json",
  "include": ["src/**/*"],
  "exclude": ["node_modules/*", "__sapper__/*", "public/*"],
  "lib": ["DOM"],
  "compilerOptions": {
    "types": ["node", "svelte"],
    /* Strict Type-Checking Options */
    "strict": true,                           /* Enable all strict type-checking options. */
    "noImplicitAny": true,                 /* Raise error on expressions and declarations with an implied 'any' type. */
    "strictNullChecks": true,              /* Enable strict null checks. */
    "strictFunctionTypes": true,           /* Enable strict checking of function types. */
    // "strictBindCallApply": true,           /* Enable strict 'bind', 'call', and 'apply' methods on functions. */
    "strictPropertyInitialization": true,  /* Enable strict checking of property initialization in classes. */
    "noImplicitThis": true,                /* Raise error on 'this' expressions with an implied 'any' type. */
    // "alwaysStrict": true,                  /* Parse in strict mode and emit "use strict" for each source file. */

    /* Additional Checks */
    "noUnusedLocals": true,                /* Report errors on unused locals. */
    "noUnusedParameters": true,            /* Report errors on unused parameters. */
    // "noImplicitReturns": true,             /* Report error when not all code paths in function return a value. */
    // "noFallthroughCasesInSwitch": true,    /* Report errors for fallthrough cases in switch statement. */
    "sourceMap": false,
    "inlineSourceMap": true,
    "inlineSources": true
  }
}
```

**Comment by jasonlyu123** [comment]
You can check the output channel near the integrated terminal in vscode. Chose svelte in the dropdown and see if there're any errors logged. Also, if you only seeing the error in a specific file. Please provide a minimal reproduiable.

**Comment by trev-dev** [comment]
Found it. Looks like svelte-check doesn't like it if you try to override the settings in the extended tsconfig.

I want inline source maps for my project because there's no way to refer to a separate sourcemap file in Shopify.

I had set `sourceMap` to `false` in my tsconfig.json then turned on inlineSources/inlineSourceMap.

This throws svelte-check for a loop because it only seems to see sourceMap: true in `@tsconfig/svelte/tsconfig.json`.

I also have my inlineSources turned on in rollup.config.js and those inline sources are indeed working. Again, more of a nuisance error. It has stopped happening now that I removed inlineSourceMap from tsconfig.json.

**Comment by jasonlyu123** [comment]
The [typescript doc](https://www.typescriptlang.org/tsconfig#inlineSourceMap) mentioned that this `inlineSourceMap` config is mutually exclusive with `sourceMap`. So as you said you would have to only enable one of them. As the base tsconfig specified it, you would have explicitly disabled it to enable `inlineSourceMap`

**Comment by trev-dev** [comment]
If you check my configs, you will see that after extending the documented, recommended tsconfig which has sourceMap: true configured inside of it, I explicitly set sourceMap: false in my own config.

It _is_ explicitly off. At least, it should be.

**Comment by jasonlyu123** [comment]
My bad. Missed that and misunderstood your comment. Found we have defaulted to enable sourceMap in the internal `svelte-preprocess` config. So that's why the error only shows on the editor not build. One workaround is to add a `svelte.config.js`:
This show disable the build-in svelte-preprocess config
```js
import sveltePreprocess from 'svelte-preprocess;

export default {
  preprocess: sveltePreprocess()
}
```
or in commonjs
```
const sveltePreprocess = require('svelte-preprocess');

module.exports = {
    preprocess: sveltePreprocess()
};
```

**Comment by jasonlyu123** [comment]
Fixed in d552d7cc45ff5e55b84032eced4875340ceb9681. Would close this when it's released.

**Comment by trev-dev** [comment]
No worries, I miss details while reading and working all the time. It's also worth mentioning again that if you leave the tsconfig alone and just pass what you want into the typescript rollup plugin, you get no errors and inline maps as well. It goes without saying it's nicer to have less confusion. Thanks for working with me on this :)
